server/registry/internal/storage/errors.go

```diff
@@ -15,15 +15,78 @@
 package storage
 
 import (
+	"context"
+	"net"
+	"os"
+
+	"github.com/apigee/registry/log"
+	"github.com/lib/pq"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+func alreadyExists(err error) bool {
+	// handle sqlite3 errors separately so their support can be conditionally compiled.
+	if isSQLite3ErrorAlreadyExists(err) {
+		return true
+	}
+	switch v := err.(type) {
+	case *pq.Error:
+		if v.Code.Name() == "unique_violation" {
+			return true
+		}
+	}
+	return false
+}
+
 // grpcErrorForDBError converts recognized database error codes to grpc error codes.
-func grpcErrorForDBError(err error) error {
-	if alreadyExists(err) {
-		return status.Error(codes.AlreadyExists, err.Error())
+func grpcErrorForDBError(ctx context.Context, err error) error {
+	// if this error already has a gRPC status code, just return it
+	if _, ok := status.FromError(err); ok {
+		return err
+	}
+	// handle sqlite3 errors separately so their support can be conditionally compiled.
+	if err2 := grpcErrorForSQLite3Error(ctx, err); err2 != nil {
+		return err2
 	}
+	// handle all other known error types.
+	switch v := err.(type) {
+	case *pq.Error:
+		if v.Code.Name() == "unique_violation" {
+			return status.Error(codes.AlreadyExists, err.Error())
```

Comment:
@theganyo This is copying the text of the original error message into the gRPC-style "Status" error that is being created (with `codes.AlreadyExists`). I think that adds the additional information that you were suggesting.